crates/ty_python_semantic/resources/mdtest/generics/legacy/functions.md

```diff
@@ -379,14 +379,13 @@ T = TypeVar("T")
 def invoke(fn: Callable[[A], B], value: A) -> B:
     return fn(value)
 
-def identity(x: T) -> T:
+def identity(x: T, /) -> T:
     return x
 
-def head(xs: list[T]) -> T:
+def head(xs: list[T], /) -> T:
     return xs[0]
 
-# TODO: this should be `Literal[1]`
-reveal_type(invoke(identity, 1))  # revealed: Unknown
+reveal_type(invoke(identity, 1))  # revealed: Literal[1]
 
 # TODO: this should be `Unknown | int`
 reveal_type(invoke(head, [1, 2, 3]))  # revealed: Unknown
```

Comment:
This TODO is not also removed because we end up inferring this constraint set when comparing `head` to `Callable[[A], B]`:

```
(B@invoke ≤ T@head) ∧ (list[T@head] ≤ A@invoke)
```

We then try to remove `T@head` from the constraint set by calculating

```
∃T@head ⋅ (B@invoke ≤ T@head) ∧ (list[T@head] ≤ A@invoke)
```

We should be able to pick `T@head = B@invoke` and simplify that to

```
(B@invoke = *) ∧ (list[B@invoke] ≤ A@invoke)
```

which I think would then be enough to propagate through the return type to discharge this TODO. I think this would require adding more derived facts to the sequent map.